### PR TITLE
Hm3 postgresql and layered architecture

### PR DESCRIPTION
**Pull request by shuspes**

### CODE REVIEW

File: src/routes/users.route.js [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
 router.route('/:userId')
-    .get(usersController.getUser)
-    .put(usersController.getUserBodyValidator(), usersController.updateUser)
-    .delete(usersController.removeUser);
+    .get(userController.getUser)
+    .put(userController.getUserBodyValidator(), userController.updateUser)
+    .delete(userController.removeUser);
 
-router.get('/:loginSubstring/:limit', usersController.getAutoSuggestUsers);
+router.get('/:loginSubstring/:limit', userController.getAutoSuggestUsers);
```
Comment (andrei-akula-epam) [inline src/routes/users.route.js:19]: I think using 'loginSubstring' and 'limit' as part of url path is not good idea.
You url looks like /users/abc/10, /users/qaz/20. But if user just wants to limit user listing? You'd better use 'loginSubstring' and 'limit' as url query parameters, in this case they can be both optional.
e.g.  
/users?sub=qaz&limit=20
/users?limit=20
/users?sub=qaz
/users

File: src/repositories/user.repository.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { Op } from 'sequelize';
+import CustomError from '../errors/customError';
+import UserModel from '../models/user';
+
+export default class LocalUserRepository {
```
Comment (andrei-akula-epam) [inline src/repositories/user.repository.js:5]: Why didn't you change the name of the class? Maybe `DbUserRepository`?

File: src/routes/users.route.js (hunk 1 above)
Comment (shuspes) [reply src/routes/users.route.js:19]: Agree, I'll make these changes

File: src/repositories/user.repository.js (hunk 2 above)
Comment (shuspes) [reply src/repositories/user.repository.js:5]: It's a typo. It should be **UserRepository** or **PgUserRepository**

**Pull request merged**